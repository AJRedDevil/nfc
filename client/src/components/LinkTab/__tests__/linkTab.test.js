// npm packages
import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';

// our packages
import {getLink, activateCurrent, LinkTab} from '../index';

describe('# LinkTab', () => {
  test('should return link with path added', () => {
    const match = {
      path: '/',
    };
    const link = {
      path: 'test',
      text: 'test',
    };
    const response = getLink({match, link});
    expect(response).toBe('/test');
  });

  test('should return link with slash and path added', () => {
    const match = {
      path: 'http://locahost:8080',
    };
    const link = {
      path: 'test',
      text: 'test',
    };
    const response = getLink({match, link});
    expect(response).toBe('http://locahost:8080/test');
  });

  test('should return active class', () => {
    const link = {
      path: 'test',
      text: 'test',
    };
    const response = activateCurrent({link, currentTab: 'test'});
    expect(response).toBe('is-active');
  });

  test('should return empty string', () => {
    const link = {
      path: 'wrong',
      text: 'test',
    };
    const response = activateCurrent({link, currentTab: 'test'});
    expect(response).toBe('');
  });

  test('LinkTab Wrapper', () => {
    const match = {
      path: 'http://locahost:8080',
    };
    const link = {
      path: 'test',
      text: 'test',
    };
    const onTabClick = () => ({message: 'test'});
    const component = renderer.create(
      <MemoryRouter location="http://locahost:8080">
        <LinkTab
          link={link}
          onTabClick={onTabClick}
          currentTab="test"
          match={match}
        />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
