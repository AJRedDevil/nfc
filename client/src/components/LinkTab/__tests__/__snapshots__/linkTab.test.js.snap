// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# LinkTab LinkTab Wrapper 1`] = `
<li
  className="is-active"
>
  <a
    aria-current={null}
    className={undefined}
    href="/http://locahost:8080/test"
    identifier="test"
    onClick={[Function]}
    style={undefined}
  >
    test
  </a>
</li>
`;
